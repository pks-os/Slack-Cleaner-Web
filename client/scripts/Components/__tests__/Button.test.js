import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';

describe('<Button />', () => {
  it('Renders Correctly', () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Renders an A element when required', () => {
    const TEXT = 'CLICK ME';
    const button = renderer.create(<Button text={TEXT} isLink href="#" />)
      .toJSON;
    expect(button).toMatchSnapshot();
  });

  it('Does not render an A element if href prop is not provided', () => {
    const TEXT = 'CLICK ME';
    const button = renderer.create(<Button text={TEXT} isLink />);
    expect(button).toMatchSnapshot();
  });
});
