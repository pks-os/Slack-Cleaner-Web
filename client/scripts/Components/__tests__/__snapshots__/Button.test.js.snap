// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Does not render an A element if href prop is not provided 1`] = `
<button
  className="Button"
  disabled={false}
  onClick={[Function]}
>
  CLICK ME
</button>
`;

exports[`<Button /> Renders Correctly 1`] = `
<button
  className="Button"
  disabled={false}
  onClick={[Function]}
>
  
</button>
`;

exports[`<Button /> Renders an A element when required 1`] = `[Function]`;
